contracts/colony/ColonyDataTypes.sol

```diff
@@ -318,10 +318,12 @@ interface ColonyDataTypes {
   struct Domain {
     uint256 skillId;
     uint256 fundingPotId;
+    bool deprecated;
   }
 
   struct LocalSkill {
     bool exists;
+    bool deprecated;
```

Comment:
I don't understand why both of these need to be added. Can we not just deprecate a domain by deprecating the associated skill?

Reply:
Currently creating a domain does not add an entry to the `LocalSkill`s mapping -- they are parallel mappings.

Reply:
I guess I don't understand what we're doing here then. It sounds like there are two totally distinct pieces of work here: 

* Adding `deprecated` to the domain mapping, which we use to deprecate domains (NB Regardless of implementation, this is the functionality that has to be restored, and what we actually care about).
* Moving tracking of local skills, and whether they are deprecated, to colonies rather than being on the network, with the intention of making things simpler and less gas-intensive.

When you were insistent on point 2 in our original discussions, I assumed it would have an effect on the implementation of point 1, and so together they would be more elegant and consistent. But that doesn't appear to be the case? If this is the sort of implementation you had in mind, why are they bundled together in one PR?

> Currently creating a domain does not add an entry to the LocalSkills mapping

Is there a reason we can't have 'backwards compatability' for such domains, in the same sort of way that you've implemented for e.g. local skill deprecation?

Reply:
It's a fair point. In my mind I was looking at "how do we improve our handling of deprecation, given its somewhat random history". Now that it's done, we could conceivably split off the domain part from the rest, but that would keep deprecation fractured. This PR makes deprecation consistent and future proofed for the network. If you'd like me to split the domain piece into a separate PR I can do that, but I think we're best taking it as a bundle.